src/us/kbase/workspace/database/provenance/Title.java

```diff
@@ -0,0 +1,229 @@
+		Title other = (Title) obj;
+		return Objects.equals(titleLanguage, other.titleLanguage)
+				&& Objects.equals(titleString, other.titleString)
+				&& Objects.equals(titleType, other.titleType);
+	}
+
+	/**
+	 * Gets a builder for an {@link Title}.
+	 *
+	 * @return the builder.
+	 */
+	public static Builder getBuilder(final String titleString) {
+		return new Builder(titleString);
+	}
+
+	/** A builder for an {@link Title}. */
+	public static class Builder {
+
+		private String titleString;
+		private TitleType titleType = null;
+		private String titleLanguage = null;
+		private List<String> errorList = new ArrayList<>();
+
+		private Builder(final String titleString) {
+			try {
+				this.titleString = Util.checkString(titleString, "titleString");
+			} catch (Exception e) {
+				this.errorList.add(e.getMessage());
+			}
+		}
+
+		/**
+		 * Sets the language of title being represented, for example en_GB.
+		 *
+		 * @param titleLanguage
+		 *                the language. Null or the empty string removes any
+		 *                current value in the builder.
+		 * @return this builder.
+		 */
+		public Builder withTitleLanguage(final String titleLanguage) {
+			this.titleLanguage = Common.processString(titleLanguage);
+			return this;
+		}
+
+		/**
+		 * Sets the type of the title being represented, for example translated_title.
+		 *
+		 * @param titleType
+		 *                the titleType as a string. Null or the empty string resets
+		 *                the current titleType in the builder.
+		 * @return this builder.
+		 */
+		public Builder withTitleType(final String titleType) {
+			final String protoTitleType = Common.processString(titleType);
+			if (protoTitleType == null) {
+				this.titleType = null;
+			} else {
+				try {
+					this.titleType = TitleType.getTitleType(protoTitleType);
+				} catch (IllegalArgumentException e) {
+					this.errorList.add(e.getMessage());
+				}
+			}
+
+			return this;
+		}
+
+		/**
+		 * Sets the type of the title being represented, for example translated_title.
+		 *
+		 * @param titleType
+		 *                the titleType as an enum. Null resets any
+		 *                current titleType in the builder.
+		 * @return this builder.
+		 */
+		public Builder withTitleType(final TitleType titleType) {
+			this.titleType = titleType;
+			return this;
+		}
+
+		/**
+		 * Build the {@link Title}.
+		 *
+		 * The titleLanguage value is checked for conformance to IETF BCP-47 syntax
+		 * rules; note that Java does not check the values themselves, but rather the
+		 * syntax of the value.
+		 *
+		 * If the titleType is null, it is set to the default DEFAULT_TITLE_TYPE.
+		 *
+		 * @return the title object.
+		 */
+		public Title build() {
+
+			// check and reformat the titleLanguage
+			if (titleLanguage != null) {
+				titleLanguage = titleLanguage.replace("_", "-");
+				Locale.Builder localeBuilder = null;
+				try {
+					localeBuilder = new Locale.Builder().setLanguageTag(titleLanguage);
```

Comment:
Try using the supplied language tag to build a locale. Java checks the syntax of the tag but doesn't check the values themselves.